### PR TITLE
Refactored `InitializeSVGS()` to fix Spotbugs error `WMI_WRONG_MAP_ITERATOR` and added test

### PR DESCRIPTION
**Pull request by StefanSpieker**
Refactored `InitializeSVGS()` to fix Spotbugs error `WMI_WRONG_MAP_ITERATOR` and added a small test to actually do something like TDD and check that at least the number of Icons is still the same. I used `greaterThanOrEqualTo` so that it does not fail if someone adds more icons.

I thought about removing the test and the `size()` method, but it might be useful for someone else also. But I can also remove that if it does not provide any value.

The refactoring will not be a performance improvement, because the Map currently just consists of one entry, but it might get, if it gets extended.

See [JENKINS-XXXXX](https://issues.jenkins-ci.org/browse/JENKINS-XXXXX).

### Proposed changelog entries

* N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: core/src/main/java/org/jenkins/ui/icon/IconSet.java
Comment (timja) [inline core/src/main/java/org/jenkins/ui/icon/IconSet.java:116]: I don’t see a reason for this to be in the public API, to me the test doesn’t add value, you could check a particular icon is in there maybe?

**Comment by timja** [comment]
Build is failing for linting

File: core/src/main/java/org/jenkins/ui/icon/IconSet.java
Comment (StefanSpieker) [reply core/src/main/java/org/jenkins/ui/icon/IconSet.java]: You are right, I just wanted to check the coreIcons size, so I removed that again, since it is already accessible.

**Comment by StefanSpieker** [comment]
> 
> 
> Build is failing for linting

Done.

**Review by timja — APPROVED** [review]

**Review by basil — APPROVED** [review]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

**Pull request merged**
